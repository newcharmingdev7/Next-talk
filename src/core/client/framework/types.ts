// TODO: (@cvle) Extract useful common types into its own package.
export { Omit, Overwrite, PropTypesOf } from "coral-ui/types";
export { DeepPartial } from "coral-common/types";

export type RelayEnumLiteral<T> = keyof T | "%future added value";
