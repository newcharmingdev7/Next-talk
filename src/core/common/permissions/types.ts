export type UserRole =
  | "%future added value"
  | "COMMENTER"
  | "MEMBER"
  | "STAFF"
  | "MODERATOR"
  | "ADMIN";

export interface User {
  id: string;
  role: UserRole;
  moderationScopes?: null | {
    scoped?: boolean;
    siteIDs?: string[];
  };
}

export type Moderator = User & {
  role: "MODERATOR";
};

export type LTEModerator = User & {
  role: "MODERATOR" | "STAFF" | "MEMBER" | "COMMENTER";
};

export const isModerator = (user: User): user is Moderator => {
  return user.role === "MODERATOR";
};

export const isLTEModerator = (user: User): user is LTEModerator => {
  return user.role !== "ADMIN";
};
