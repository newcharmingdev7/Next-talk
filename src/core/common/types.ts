export type Diff<T extends keyof any, U extends keyof any> = ({ [P in T]: P } &
  { [P in U]: never } & { [x: string]: never })[T];

export type Omit<U, K extends keyof U> = Pick<U, Diff<keyof U, K>>;

export type Overwrite<T, U> = Pick<T, Diff<keyof T, keyof U>> & U;

export type Sub<T, U> = Pick<T, Diff<keyof T, keyof U>>;

/**
 * Make all properties in T writeable
 */
export type Writeable<T> = { -readonly [P in keyof T]: T[P] };

export type Promiseable<T> = Promise<T> | T;
