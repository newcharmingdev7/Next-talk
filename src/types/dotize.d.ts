declare module "dotize" {
  export function convert(obj: Object): Record<string, any>
}
