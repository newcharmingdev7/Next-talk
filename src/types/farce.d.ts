declare module "farce" {
  export class BrowserProtocol {}
  export class queryMiddleware {}
}
