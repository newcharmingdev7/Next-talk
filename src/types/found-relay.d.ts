declare module "found-relay" {
  import { Environment } from "relay-runtime";
  export class Resolver {
    constructor(relayEnvironment: Environment);
  }
}
