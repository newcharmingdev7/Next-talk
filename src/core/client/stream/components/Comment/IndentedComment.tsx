import React, { StatelessComponent } from "react";

import { PropTypesOf } from "talk-framework/types";

import Indent from "../Indent";
import Comment from "./Comment";

export interface IndentedCommentProps extends PropTypesOf<typeof Comment> {
  indentLevel?: number;
}

const IndentedComment: StatelessComponent<IndentedCommentProps> = props => {
  const { indentLevel, ...rest } = props;
  const CommentElement = <Comment {...rest} />;
  const CommentwithIndent =
    (indentLevel && <Indent level={indentLevel}>{CommentElement}</Indent>) ||
    CommentElement;
  return CommentwithIndent;
};

export default IndentedComment;
