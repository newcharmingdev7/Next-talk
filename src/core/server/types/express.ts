import { Request } from "express";

import { Tenant } from "talk-server/models/tenant";
import { User } from "talk-server/models/user";
import TenantCache from "talk-server/services/tenant/cache";

export interface Request extends Request {
  user?: User;
  tenant?: Tenant;
  tenantCache: TenantCache;
}
