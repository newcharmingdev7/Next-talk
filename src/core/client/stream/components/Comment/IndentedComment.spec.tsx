import { shallow } from "enzyme";
import React from "react";

import { PropTypesOf } from "talk-framework/types";

import IndentedComment from "./IndentedComment";

it("renders correctly", () => {
  const props: PropTypesOf<typeof IndentedComment> = {
    indentLevel: 1,
    id: "comment-id",
    author: {
      username: "Marvin",
    },
    body: "Woof",
    createdAt: "1995-12-17T03:24:00.000Z",
  };
  const wrapper = shallow(<IndentedComment {...props} />);
  expect(wrapper).toMatchSnapshot();
});
