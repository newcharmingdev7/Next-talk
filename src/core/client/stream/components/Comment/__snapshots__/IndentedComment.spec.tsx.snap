// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Indent
  level={1}
>
  <Comment
    author={
      Object {
        "username": "Marvin",
      }
    }
    body="Woof"
    createdAt="1995-12-17T03:24:00.000Z"
    id="comment-id"
  />
</Indent>
`;
